scripts/globals/npc_util.lua

```diff
@@ -396,8 +396,20 @@ function npcUtil.completeQuest(player, area, quest, params)
         end
     end
 
+    local logId
+    if type(area) == "number" then
+        logId = area
+    elseif area["quest_log"] then
+        logId = area["quest_log"]
+    end
+
```

Comment:
This is something I need to revisit in the future.  The ultimate goal is to get away from using the log_ids table, and switch to using tpz.quest.log_id.? for quest related items.  For now though, we're still dependent on it due to fame areas.